apple/OmnivoreKit/Sources/Services/AudioSession/AudioSession.swift

```diff
@@ -19,24 +19,35 @@ public enum AudioSessionState {
   case playing
 }
 
+public enum PlayerScrubState {
+  case reset
+  case scrubStarted
+  case scrubEnded(TimeInterval)
+}
+
 // Our observable object class
-public class AudioSession: ObservableObject {
+public class AudioSession: NSObject, ObservableObject, AVAudioPlayerDelegate {
   @Published public var state: AudioSessionState = .stopped
   @Published public var item: LinkedItem?
 
+  @Published public var timeElapsed: TimeInterval = 0
+  @Published public var duration: TimeInterval = 0
+  @Published public var timeElapsedString: String?
+  @Published public var durationString: String?
+
   let appEnvironment: AppEnvironment
   let networker: Networker
 
   var timer: Timer?
   var player: AVAudioPlayer?
+  var downloadTask: Task<Void, Error>?
```

Comment:
nice. better way to track cancellations.

Reply:
Yeah I really like that `Task.isCancelled` thing, so you don't even need to track state.